### ISSUE TITLE
`renv/activate.R` 1.0.8 stops processing of `.Rprofile` in RStudio

### ISSUE THREAD

**Issue by jabenninghoff** [rstudio/renv#1990]
When opening a project in RStudio (2024.04.2+764), `source("renv/activate.R")` stops processing of the `.Rprofile`.

Steps to reproduce:

1. Create a new project in RStudio, selecting "Use renv with this project"
2. Modify the `.Rprofile` as follows:

```r
writeLines("start")
writeLines("activate")
source("renv/activate.R")
writeLines("options")
options(
  warnPartialMatchArgs = TRUE,
  warnPartialMatchAttr = TRUE,
  warnPartialMatchDollar = TRUE
)
writeLines("end")
```

Starting RStudio with renv 1.0.7:

```
Restarting R session...

start
activate
- Project '~/GitHub/renv-test' loaded. [renv 1.0.7]
options
end
```

Starting RStudio with renv 1.0.8:

```
Restarting R session...

start
activate
- Project '~/GitHub/renv-test' loaded. [renv 1.0.8]
```

Starting R with renv 1.0.8:

```
start
activate
- Project '~/GitHub/renv-test' loaded. [renv 1.0.8]
options
end
```

This seems to be caused by #1915.

**Comment by salim-b** [comment]
I ran into the same issue after upgrading to renv 1.0.8. This is a serious regression!

I don't have a clear enough view of the renv starting behavior to be able to say exactly where things go south, but I noticed that setting `options(renv.config.autoloader.enabled = FALSE)` makes the `.Rprofile` script being executed fully again – with the downside of not properly loading the renv package cage anymore.

**Comment by kevinushey** [comment]
Thanks for the bug report -- we'll take a look and try to get this fixed up ASAP.

**Comment by kevinushey** [comment]
I've pushed a potential fix in https://github.com/rstudio/renv/commit/73d3841273690e8f094f208cca3802123d520e07 that works locally for me -- are you able to test and report back? If everything looks good I can prepare a patch release for CRAN soon.

**Comment by jabenninghoff** [comment]
@kevinushey I have tested the patched version locally and it does resolve the issue for me. Thanks!

**Comment by salim-b** [comment]
@kevinushey I can also confirm that this issue is fixed with `renv::upgrade(version = "73d3841")`.

**Comment by kevinushey** [comment]
Thanks for testing and confirming! renv 1.0.9 is now on CRAN; please let me know if you encounter any other issues.
